Cargo.lock

```diff
@@ -4877,17 +5226,30 @@ dependencies = [
 
 [[package]]
 name = "rustix"
-version = "0.38.4"
+version = "0.38.11"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "0a962918ea88d644592894bc6dc55acc6c0956488adcebbfb6e273506b7fd6e5"
+checksum = "c0c3dde1fc030af041adc40e79c0e7fbcf431dd24870053d187d7c66e4b87453"
 dependencies = [
- "bitflags 2.3.3",
+ "bitflags 2.4.0",
  "errno",
  "libc",
- "linux-raw-sys 0.4.3",
+ "linux-raw-sys 0.4.5",
  "windows-sys",
 ]
 
+[[package]]
+name = "rustls"
+version = "0.18.1"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "5d1126dcf58e93cee7d098dbda643b5f92ed724f1f6a63007c1116eed6700c81"
+dependencies = [
+ "base64 0.12.3",
+ "log",
+ "ring",
+ "sct 0.6.1",
+ "webpki 0.21.4",
```

Comment:
This re-introduces the same RUSTSEC, right?

https://github.com/libp2p/rust-libp2p/pull/4381

Reply:
yup, good catch. Pff it's `surf` see [here](https://github.com/http-rs/surf/issues/352) and [here](https://github.com/rustsec/advisory-db/issues/1471). There's a PR to [update it](https://github.com/http-rs/surf/pull/340) to `0.19` but no response. Besides `surf` I can't find another http client compatible with `async-std`, more and more feels like https://github.com/libp2p/rust-libp2p/discussions/4286.

Reply:
I am tempted to say lets make the first step towards removing `async-std` by not including it here. We can then do https://github.com/libp2p/rust-libp2p/issues/4449 next and once that is complete, phase it out completely.

Thoughts?

Reply:
:+1: for shipping this pull request without async-std. I am sorry for the trouble you went through to make this multi-runtime @jxs!

Reply:
no worries Max :) 
Removed `async-std` support, the CI failing seems unrelated to this PR, regarding `cargo-audit` I reopened https://github.com/libp2p/rust-libp2p/issues/4375#issuecomment-1710215558